pkg/sqlite/migrations/007_replaces_skips.go

```diff
@@ -0,0 +1,116 @@
+var replacesSkipsMigration = &Migration{
+	Id: ReplacesSkipsMigrationKey,
+	Up: func(ctx context.Context, tx *sql.Tx) error {
+		sql := `
+		ALTER TABLE operatorbundle 
+		ADD COLUMN replaces TEXT;
+
+		ALTER TABLE operatorbundle 
+		ADD COLUMN skips TEXT;
+		
+		CREATE TRIGGER api_provider_cleanup
+		AFTER DELETE ON api_provider
+		WHEN NOT EXISTS (SELECT 1 FROM api_provider JOIN api_requirer WHERE 
+		    (api_provider.group_name = OLD.group_name AND api_provider.version = OLD.version AND api_provider.kind = OLD.kind) OR 
+			(api_requirer.group_name = OLD.group_name AND api_requirer.version = OLD.version AND api_requirer.kind = OLD.kind))
+		BEGIN
+			DELETE FROM api WHERE group_name = OLD.group_name AND version = OLD.version AND kind = OLD.kind;
+		END;
+		
+		CREATE TRIGGER api_requirer_cleanup
+		AFTER DELETE ON api_requirer
+		WHEN NOT EXISTS (SELECT 1 FROM api_provider JOIN api_requirer WHERE 
+		    (api_provider.group_name = OLD.group_name AND api_provider.version = OLD.version AND api_provider.kind = OLD.kind) OR 
+			(api_requirer.group_name = OLD.group_name AND api_requirer.version = OLD.version AND api_requirer.kind = OLD.kind))	
+		BEGIN
+			DELETE FROM api WHERE group_name = OLD.group_name AND version = OLD.version AND kind = OLD.kind;
+		END;
+		`
+		_, err := tx.ExecContext(ctx, sql)
+		if err != nil {
+			return err
+		}
+
+		bundles, err := listBundles(ctx, tx)
+		if err != nil {
+			return err
+		}
+		for _, bundle := range bundles {
+			if err := extractReplaces(ctx, tx, bundle); err != nil {
+				logrus.Warnf("error backfilling replaces and skips: %v", err)
+				continue
+			}
+		}
+		return err
+	},
+	Down: func(ctx context.Context, tx *sql.Tx) error {
+		foreignKeyOff := `PRAGMA foreign_keys = 0`
+		createTempTable := `CREATE TABLE operatorbundle_backup (name TEXT, csv TEXT, bundle TEXT, bundlepath TEXT, version TEXT, skiprange TEXT)`
+		backupTargetTable := `INSERT INTO operatorbundle_backup SELECT name, csv, bundle, bundlepath, version, skiprange FROM operatorbundle`
+		dropTargetTable := `DROP TABLE operatorbundle`
+		renameBackUpTable := `ALTER TABLE operatorbundle_backup RENAME TO operatorbundle;`
+		foreignKeyOn := `PRAGMA foreign_keys = 1`
+		_, err := tx.ExecContext(ctx, foreignKeyOff)
+		if err != nil {
+			return err
+		}
+		_, err = tx.ExecContext(ctx, createTempTable)
+		if err != nil {
+			return err
+		}
+		_, err = tx.ExecContext(ctx, backupTargetTable)
+		if err != nil {
+			return err
+		}
+		_, err = tx.ExecContext(ctx, dropTargetTable)
+		if err != nil {
+			return err
+		}
+		_, err = tx.ExecContext(ctx, renameBackUpTable)
+		if err != nil {
+			return err
+		}
+		_, err = tx.ExecContext(ctx, foreignKeyOn)
+		return err
+	},
+}
+
+func extractReplaces(ctx context.Context, tx *sql.Tx, name string) error {
+	addSql := `insert into operatorbundle(replaces, skips) values(?,?)`
+
+	// TODO: what should we do when there are bundle images?
```

Comment:
Can we infer replaces and skips from the channel_entries table?

Reply:
Discussed offline, but persisting here. I was thinking we couldn't distinguish between replaces and skips in the channel entry table, but after some discussion we should be able to distinguish between them based on depth.